<a href="https://colab.research.google.com/github/Eunjung-Cho/python_study/blob/master/API/OpenDartReader_unlisting_corps.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# OpenDartReader - 비상장 기업 데이터 조회와 활용

<img width="40%" src="https://i.imgur.com/NE65l5L.png" >

***2019-2022 [FinanceData.KR]()***

## 요약

```python
import OpenDartReader

api_key = '961aebb762ef44e8f93a36218897eb**********'
dart = OpenDartReader(api_key)

# [비바리퍼블리카] (송금서비스 토스로 유명한), 공시(감사보고서) 목록
df = dart.list('비바리퍼블리카', start='1900')

# [빅히트엔터테인먼트] (BTS로 유명한), 공시(감사보고서) 목록
df = dart.list('빅히트엔터테인먼트', start='1900')
df = dart.list('01204056', start='1900')

# [쿠팡] "연결감사보고서 (2019.12)" 20200414000766 (삼일회계법인)
# http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20200414000766
html_text = dart.document('20200414000766')
df_list = pd.read_html(html_text)
df_list[15] # 연결포괄손익계산서
```

<img src="https://i.imgur.com/c3t2toM.png" >


## OpenDartReader 설치와 초기화

In [ ]:
!pip install -q OpenDartReader

In [ ]:
import OpenDartReader

# 발급받은 API키를 사용하세요
api_key = 'c19d19f2fae8006979b6e3670c8e8afe9d7d44cf'

dart = OpenDartReader(api_key)

## 비상장 회사의 공시보고서 목록

In [ ]:
df = dart.list('비바리퍼블리카', start='1900') # 토스
df

corp_code corp_name stock_code corp_cls          report_nm        rcept_no  \
0   01212921   비바리퍼블리카                   E    주요사항보고서(유상증자결정)  20221020000382   
1   01212921   비바리퍼블리카                   E    주요사항보고서(유상증자결정)  20221018000362   
2   01212921   비바리퍼블리카                   E    반기보고서 (2022.06)  20220829000672   
3   01212921   비바리퍼블리카                   E    주요사항보고서(유상증자결정)  20220825000567   
4   01212921   비바리퍼블리카                   E    주요사항보고서(유상증자결정)  20220720000316   
5   01212921   비바리퍼블리카                   E    분기보고서 (2022.03)  20220530000844   
6   01212921   비바리퍼블리카                   E    사업보고서 (2021.12)  20220331003427   
7   01212921   비바리퍼블리카                   E  연결감사보고서 (2020.12)  20210406000710   
8   01212921   비바리퍼블리카                   E    감사보고서 (2020.12)  20210406000707   
9   01212921   비바리퍼블리카                   E    감사보고서 (2019.12)  20200410003129   
10  01212921   비바리퍼블리카                   E    감사보고서 (2018.12)  20190412002776   
11  01212921   비바리퍼블리카                   E    감사보고서 (2017.12)  20180413001411   

     flr_nm  rcept_dt rm  
0   비바리퍼블리카  20221020     
1   비바리퍼블리카  20221018     
2   비바리퍼블리카  20220829     
3   비바리퍼블리카  20220825     
4   비바리퍼블리카  20220720     
5   비바리퍼블리카  20220530     
6   비바리퍼블리카  20220331  연  
7    삼일회계법인  20210406     
8    삼일회계법인  20210406     
9    삼일회계법인  20200410     
10   삼일회계법인  20190412     
11   삼일회계법인  20180413

In [ ]:
# 보고서 링크
for ix, row in df.iterrows():
    rcp_no = row['rcept_no']
    print(f'http://dart.fss.or.kr/dsaf001/main.do?rcpNo={rcp_no}')

http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20221020000382
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20221018000362
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220829000672
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220825000567
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220720000316
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220530000844
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220331003427
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20210406000710
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20210406000707
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20200410003129
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20190412002776
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20180413001411


## 이름으로 확인 하기

회사 이름으로 식별이 고유하게 식별이 가능한지 확인해야 하는 것을 추천합니다.

In [ ]:
comps = dart.company_by_name('하이브')
len(comps)

22

In [ ]:
comps

[{'status': '000',
  'message': '정상',
  'corp_code': '00797373',
  'corp_name': '마포하이브로드파킹(주)',
  'corp_name_eng': 'Mapo Hibroad Parking co., Ltd.',
  'stock_name': '마포하이브로드파킹',
  'stock_code': '',
  'ceo_nm': '도기훈',
  'corp_cls': 'E',
  'jurir_no': '1101114250140',
  'bizr_no': '2208792901',
  'adres': '인천광역시 연수구 인천타워대로 241 24층(송도동,포스코건설사옥)',
  'hm_url': '',
  'ir_url': '',
  'phn_no': '032-748-2754',
  'fax_no': '032-748-2754',
  'induty_code': '41229',
  'est_dt': '20091228',
  'acc_mt': '12'},
 {'status': '000',
  'message': '정상',
  'corp_code': '00537823',
  'corp_name': '하이브랜드 에이비씨피주식회사',
  'corp_name_eng': 'Hibrand ABCP Ltd.',
  'stock_name': '하이브랜드에이비씨피',
  'stock_code': '',
  'ceo_nm': '신언도',
  'corp_cls': 'E',
  'jurir_no': '1101113240283',
  'bizr_no': '2148772045',
  'adres': '서울특별시 서초구 서초동 1361-9 서광빌딩 9층',
  'hm_url': '',
  'ir_url': '',
  'phn_no': '02-6263-8219',
  'fax_no': '02-6260-2345',
  'induty_code': '64999',
  'est_dt': '20050601',
  'acc_mt': '03'},
 {'status': 

## 공시목록 가져오기

조회를 해보고 정식명칭의 전체 회사명을 사용하십시오. `corp_code`를 쓰는 것이 가장 확실합니다.

In [ ]:
bighit = dart.list('하이브', start='1900')
bighit

corp_code corp_name stock_code corp_cls            report_nm  \
0    01204056       하이브     352820        Y         투자판단관련주요경영사항   
1    01204056       하이브     352820        Y    기업설명회(IR)개최(안내공시)   
2    01204056       하이브     352820        Y    주식등의대량보유상황보고서(일반)   
3    01204056       하이브     352820        Y       주식매수선택권부여에관한신고   
4    01204056       하이브     352820        Y  영업양수결정(종속회사의주요경영사항)   
..        ...       ...        ...      ...                  ...   
129  01204056       하이브     352820        Y  [발행조건확정]증권신고서(지분증권)   
130  01204056       하이브     352820        Y    연결감사보고서 (2019.12)   
131  01204056       하이브     352820        Y      감사보고서 (2019.12)   
132  01204056       하이브     352820        Y      감사보고서 (2018.12)   
133  01204056       하이브     352820        Y      감사보고서 (2017.12)   

           rcept_no  flr_nm  rcept_dt rm  
0    20221017800268     하이브  20221017  유  
1    20221014800412     하이브  20221014  유  
2    20221014000379     방시혁  20221014     
3    20220927000446     하이브  20220927     
4    20220830800533     하이브  20220830  유  
..              ...     ...       ... ..  
129  20200928000122     하이브  20200928     
130  20200331000038  한영회계법인  20200331     
131  20200331000037  한영회계법인  20200331     
132  20190319000209  삼일회계법인  20190319     
133  20180322000053  삼일회계법인  20180322     

[134 rows x 9 columns]

In [ ]:
bighit = dart.list('01204056', start='1900')
bighit

corp_code corp_name stock_code corp_cls            report_nm  \
0    01204056       하이브     352820        Y         투자판단관련주요경영사항   
1    01204056       하이브     352820        Y    기업설명회(IR)개최(안내공시)   
2    01204056       하이브     352820        Y    주식등의대량보유상황보고서(일반)   
3    01204056       하이브     352820        Y       주식매수선택권부여에관한신고   
4    01204056       하이브     352820        Y  영업양수결정(종속회사의주요경영사항)   
..        ...       ...        ...      ...                  ...   
129  01204056       하이브     352820        Y  [발행조건확정]증권신고서(지분증권)   
130  01204056       하이브     352820        Y    연결감사보고서 (2019.12)   
131  01204056       하이브     352820        Y      감사보고서 (2019.12)   
132  01204056       하이브     352820        Y      감사보고서 (2018.12)   
133  01204056       하이브     352820        Y      감사보고서 (2017.12)   

           rcept_no  flr_nm  rcept_dt rm  
0    20221017800268     하이브  20221017  유  
1    20221014800412     하이브  20221014  유  
2    20221014000379     방시혁  20221014     
3    20220927000446     하이브  20220927     
4    20220830800533     하이브  20220830  유  
..              ...     ...       ... ..  
129  20200928000122     하이브  20200928     
130  20200331000038  한영회계법인  20200331     
131  20200331000037  한영회계법인  20200331     
132  20190319000209  삼일회계법인  20190319     
133  20180322000053  삼일회계법인  20180322     

[134 rows x 9 columns]

## 공시 본문가져오기

"연결감사보고서 (2019.12)"의 접수번호(rcept_no) '20200331000038'로 공시보고서 문서 전체를 얻을 수 있습니다. (형식은 XHTML 입니다)

http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20200331000038




In [ ]:
html_text = dart.document('20200331000038')
html_text[:1000] # 앞부분

'<?xml version="1.0" encoding="utf-8"?>\r\n\r\n\r\n<DOCUMENT xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="dart3.xsd">\n\n<DOCUMENT-HEADER AEXT-CLASS="Y">\n<DOCUMENT-NAME ACODE="00761">연결감사보고서</DOCUMENT-NAME>\n<FORMULA-VERSION SUBVER="4" ADATE="20190212">3.4</FORMULA-VERSION>\n<COMPANY-NAME AREGCIK="01204056" AACCOUNTTYPE="A">주식회사 빅히트엔터테인먼트</COMPANY-NAME>\n\n<SUMMARY>\n<EXTRACTION ACODE="AUDIT_CIK" AFEATURE="BOTH">00260198</EXTRACTION>\n<EXTRACTION ACODE="SUPV_OPIN" AFEATURE="BOTH">100000000000</EXTRACTION>\n<EXTRACTION ACODE="TOT_ASSETS" AFEATURE="BOTH">362990</EXTRACTION>\n<EXTRACTION ACODE="TOT_DEBTS" AFEATURE="BOTH">189467</EXTRACTION>\n<EXTRACTION ACODE="TOT_SALES" AFEATURE="BOTH">587224</EXTRACTION>\n<EXTRACTION ACODE="TOT_EMPL" AFEATURE="BOTH">228</EXTRACTION>\n<EXTRACTION ACODE="ETC_OTHR_YN" AFEATURE="BOTH">Y</EXTRACTION>\n<EXTRACTION ACODE="POW_OTHR_YN" AFEATURE="BOTH">N</EXTRACTION>\n<EXTRACTION ACODE="ACC_PROF_YN" AFEATURE="BOTH">Y</EXT

## 재무데이터를 데이터프레임으로 읽어오기

pandas의 read_html()로 페이지에 포함된 표를 손쉽게 DataFrame으로 읽을 수 있습니다.

In [ ]:
df_list = pd.read_html(html_text)

In [ ]:
len(df_list)

142

페이지안에 총 142개의 테이블(table)이 있습니다.

몇 번째 표가 `연결재무상태표`인지는 인덱스번호를 넣어가며 찾아야 합니다.  여기서는 15번째(인덱스로는 14)가 연결재무상태표 입니다.

In [ ]:
# 연결재무상태표
df = df_list[14]

remove_text = "&cr;(감사받지 아니한 재무제표)"
df.columns = [col.replace(remove_text, '') for col in df.columns]
df

과 목         주 석     제15(당) 기말 제15(당) 기말.1      제14(전) 기말  \
0                  자산         NaN           NaN         NaN            NaN   
1                유동자산         NaN           NaN   294652316            NaN   
2            현금및현금성자산        33.0     160780830         NaN       58064738   
3   당기손익-공정가치 측정 금융자산     43233.0             -         NaN       36817279   
4                매출채권     53033.0      94607697         NaN       49201245   
5            기타유동금융자산       633.0       5750441         NaN       12450900   
6              기타유동자산         7.0      20541758         NaN        9468209   
7                재고자산         8.0      12971590         NaN        5705147   
8               비유동자산         NaN           NaN    68337520            NaN   
9   당기손익-공정가치 측정 금융자산     43233.0       1572021         NaN        1892865   
10          기타비유동금융자산       633.0      10757572         NaN        3194526   
11             공동기업투자         9.0       2327780         NaN        3235619   
12               유형자산        10.0       8184913         NaN        8751498   
13              사용권자산        34.0       8654524         NaN        9434846   
14               무형자산        11.0      25743454         NaN        6328275   
15            이연법인세자산        18.0       8523282         NaN         458811   
16            기타비유동자산         7.0       2573974         NaN         525210   
17              자산 총계         NaN           NaN   362989836            NaN   
18                 부채         NaN           NaN         NaN            NaN   
19               유동부채         NaN           NaN   174561335            NaN   
20               매입채무      3133.0      25342680         NaN       13575752   
21          전환상환우선주부채    143133.0       5143282         NaN        9301013   
22               리스부채    313334.0       5289457         NaN        2959831   
23           기타유동금융부채  12303133.0      36931276         NaN       33893664   
24            당기법인세부채         NaN      15946850         NaN       13264223   
25             기타유동부채        13.0      84828925         NaN       32436298   
26            유동성장기부채        15.0       1010397         NaN              -   
27               충당부채        16.0         68468         NaN          31557   
28              비유동부채         NaN           NaN    14905580            NaN   
29          전환상환우선주부채    143133.0             -         NaN              -   
30               리스부채    313334.0       2841429         NaN        4878439   
31              장기차입금        15.0        209790         NaN              -   
32          기타비유동금융부채    123033.0          3000         NaN              -   
33            순확정급여부채        17.0       4285728         NaN        2990303   
34            이연법인세부채        18.0         73294         NaN         593707   
35            기타비유동부채        13.0       5772378         NaN        2592956   
36               충당부채        16.0       1719961         NaN        1751891   
37              부채 총계         NaN           NaN   189466915            NaN   
38                 자본         NaN           NaN         NaN            NaN   
39         지배기업의 소유지분         NaN           NaN   174183579            NaN   
40                자본금       119.0        834881         NaN         772917   
41            주식발행초과금        19.0     211100917         NaN      198574802   
42               기타자본        20.0       4025286         NaN        3522288   
43          기타포괄손익누계액        20.0      (57,982)         NaN       (67,888)   
44                결손금         NaN  (41,719,523)         NaN  (115,542,585)   
45              비지배지분         NaN           NaN   (660,658)              -   
46              자본 총계         NaN           NaN   173522921            NaN   
47          부채와 자본 총계         NaN           NaN   362989836            NaN   

    제14(전) 기말.1     제14(전) 기초   제14(전) 기초.1  
0           NaN           NaN           NaN  
1   171707518.0           NaN      55866023  
2           NaN      29504093           NaN  
3          

In [ ]:
# 연결포괄손익계산서
df = df_list[18]

remove_text = "&cr;(감사받지 아니한 재무제표)"
df.columns = [col.replace(remove_text, '') for col in df.columns]
df

과 목     주 석       제15기       제15기.1       제14기  \
0                        매출액  2230.0        NaN    587224478        NaN   
1                       매출원가  2430.0        NaN    385686888        NaN   
2                      매출총이익     NaN        NaN    201537590        NaN   
3                    판매비와관리비  2324.0  102795121          NaN   31462308   
4                       영업이익     NaN        NaN     98742469        NaN   
5                       기타수익    25.0     154231          NaN      97511   
6                       기타비용    25.0     510638          NaN    1768914   
7                      지분법손실     9.0     907839          NaN     124381   
8                       금융수익  2633.0    7490975          NaN    2207145   
9                       금융비용  2633.0    6476775          NaN  128791219   
10               법인세비용차감전순손익     NaN        NaN     98492423        NaN   
11                     법인세비용    18.0        NaN     26068328        NaN   
12                     당기순손익     NaN        NaN     72424095        NaN   
13        지배기업의 소유주 귀속 당기순이익     NaN        NaN     73823063        NaN   
14            비지배지분 귀속 당기순이익     NaN        NaN  (1,398,968)        NaN   
15            법인세비용차감후기타포괄손익     NaN        NaN        11384        NaN   
16           당기손익으로 재분류되는 항목     NaN        NaN       319572        NaN   
17                  해외사업환산차이     NaN     319572          NaN     149134   
18  후속적으로 당기손익으로 재분류되지 않는 항목     NaN        NaN    (308,188)        NaN   
19            순확정급여부채의 재측정요소     NaN  (308,188)          NaN  (203,247)   
20                     총포괄손익     NaN        NaN     72435479        NaN   
21        지배기업의 소유주 귀속 총포괄손익     NaN        NaN     73832969        NaN   
22            비지배지분 귀속 총포괄손익     NaN        NaN  (1,397,490)        NaN   
23              주당순이익 (단위:원)     NaN        NaN          NaN        NaN   
24                   기본주당순손익    27.0        NaN        47219        NaN   
25                   희석주당순손익    27.0        NaN        42174        NaN   

          제14기.1  
0      301371824  
1      189978918  
2      111392906  
3            NaN  
4       79930598  
5            NaN  
6            NaN  
7            NaN  
8            NaN  
9            NaN  
10  (48,449,260)  
11      22016319  
12  (70,465,579)  
13  (70,465,579)  
14             -  
15      (54,113)  
16        149134  
17           NaN  
18     (203,247)  
19           NaN  
20  (70,519,692)  
21  (70,519,692)  
22             -  
23           NaN  
24      (54,030)  
25      (54,030)

In [ ]:
# 연결현금흐름표
df = df_list[24]

remove_text = "&cr;(감사받지 아니한 재무제표)"
df.columns = [col.replace(remove_text, '') for col in df.columns]
df

구 분   주 석          제15기       제15기.1          제14기  \
0          영업활동으로 인한 현금흐름   NaN           NaN     91713636           NaN   
1         영업으로부터 창출된 현금흐름  28.0     121917870          NaN      88169367   
2                  이자의 수취   NaN        175611          NaN         89461   
3                  이자의 지급   NaN      (12,979)          NaN         (117)   
4                 법인세의 납부   NaN  (30,366,866)          NaN  (13,571,962)   
5          투자활동으로 인한 현금흐름   NaN           NaN     12180077           NaN   
6              기타금융자산의 감소   NaN     108939949          NaN      71656309   
7           기타비유동금융자산의 감소   NaN        400019          NaN         86064   
8                유형자산의 처분   NaN         27891          NaN             -   
9       종속기업 취득에 따른 순현금흐름   NaN  (15,202,551)          NaN             -   
10      관계기업 및 공동기업투자의 취득   NaN             -          NaN   (3,360,000)   
11             기타금융자산의 증가   NaN  (65,568,271)          NaN  (95,971,454)   
12          기타비유동금융자산의 증가   NaN   (8,031,418)          NaN   (3,426,566)   
13               유형자산의 취득   NaN   (4,836,016)          NaN   (6,609,511)   
14               무형자산의 취득   NaN   (1,638,699)          NaN   (7,165,982)   
15            기타비유동자산의 증가   NaN   (1,910,827)          NaN     (276,893)   
16         재무활동으로 인한 현금흐름   NaN           NaN  (1,436,486)           NaN   
17              비지배지분의 증가   NaN       2000000          NaN             -   
18              주식선택권의 행사   NaN        170000          NaN             -   
19                차입금의 차입   NaN           800          NaN        342650   
20                차입금의 상환   NaN     (190,787)          NaN     (342,650)   
21             기타금융부채의 감소   NaN             -          NaN      (24,000)   
22               리스부채의 감소   NaN   (3,416,500)          NaN   (1,369,794)   
23           현금및현금성자산의 증가   NaN           NaN    102457227           NaN   
24  외화표시 현금및현금성자산의 환율변동효과   NaN           NaN       258865           NaN   
25            기초 현금및현금성자산   NaN           NaN     58064738           NaN   
26            기말 현금및현금성자산   NaN           NaN    160780830           NaN   

          제14기.1  
0       74686749  
1            NaN  
2            NaN  
3            NaN  
4            NaN  
5   (45,068,033)  
6            NaN  
7            NaN  
8            NaN  
9            NaN  
10           NaN  
11           NaN  
12           NaN  
13           NaN  
14           NaN  
15           NaN  
16   (1,393,794)  
17           NaN  
18           NaN  
19           NaN  
20           NaN  
21           NaN  
22           NaN  
23      28224922  
24        335723  
25      29504093  
26      58064738

## 쿠팡 손익계산서

In [ ]:
coupang = dart.list('쿠팡', start='1900')
coupang

corp_code corp_name stock_code corp_cls  \
0   01019166        쿠팡                   E   
1   01019166        쿠팡                   E   
2   01019166        쿠팡                   E   
3   01019166        쿠팡                   E   
4   01019166        쿠팡                   E   
5   01019166        쿠팡                   E   
6   01019166        쿠팡                   E   
7   01019166        쿠팡                   E   
8   01019166        쿠팡                   E   
9   01019166        쿠팡                   E   
10  01019166        쿠팡                   E   
11  01019166        쿠팡                   E   
12  01019166        쿠팡                   E   
13  01019166        쿠팡                   E   
14  01019166        쿠팡                   E   
15  01019166        쿠팡                   E   
16  01019166        쿠팡                   E   
17  01019166        쿠팡                   E   
18  01019166        쿠팡                   E   
19  01019166        쿠팡                   E   
20  01019166        쿠팡                   E   
21  01019166        쿠팡                   E   
22  01019166        쿠팡                   E   
23  01019166        쿠팡                   E   
24  01019166        쿠팡                   E   
25  01019166        쿠팡                   E   
26  01019166        쿠팡                   E   
27  01019166        쿠팡                   E   
28  01019166        쿠팡                   E   
29  01019166        쿠팡                   E   
30  01019166        쿠팡                   E   
31  01019166        쿠팡                   E   
32  01019166        쿠팡                   E   
33  01019166        쿠팡                   E   
34  01019166        쿠팡                   E   
35  01019166        쿠팡                   E   
36  01019166        쿠팡                   E   
37  01019166        쿠팡                   E   
38  01019166        쿠팡                   E   
39  01019166        쿠팡                   E   
40  01019166        쿠팡                   E   
41  01019166        쿠팡                   E   
42  01019166        쿠팡                   E   
43  01019166        쿠팡                   E   
44  01019166        쿠팡                   E   
45  01019166        쿠팡                   E   
46  01019166        쿠팡                   E   
47  01019166        쿠팡                   E   
48  01019166        쿠팡                   E   
49  01019166        쿠팡                   E   
50  01019166        쿠팡                   E   
51  01019166        쿠팡                   E   

                                report_nm        rcept_no  flr_nm  rcept_dt rm  
0                                   임원의변동  20220908000618      쿠팡  20220908  공  
1                              특수관계인에대한출자  20220908000617      쿠팡  20220908  공  
2                            특수관계인에대한담보제공  20220908000616      쿠팡  20220908  공  
3               대규모기업집단현황공시[분기별공시(대표회사용)]  20220831000576      쿠팡  20220831  공  
4                             특수관계인과의내부거래  20220824000408      쿠팡  20220824  공  
5                           특수관계인에대한부동산임대  20220823000291      쿠팡  20220823  공  
6                           특수관계인에대한부동산임대  20220823000289      쿠팡  20220823  공  
7                           특수관계인에대한부동산임대  20220823000286      쿠팡  20220823  공  
8                                  유상증자결정  20220802000314      쿠팡  20220802  공  
9                           특수관계인에대한부동산임대  20220719000373      쿠팡  20220720  공  
10                          특수관계인에대한부동산임대  20220719000372      쿠팡  20220720  공  
11                          특수관계인에대한부동산임대  20220719000371      쿠팡  20220720  공  
12                           특수관계인에대한영업양도  20220630000914      쿠팡  20220630  공  
13                           특수관계인에대한자산양도  20220630000911      쿠팡  20220630  공  
14                          특수관계인에대한부동산임대  20220630000907      쿠팡  20220630  공  
15                          특수관계인에대한부동산임대  20220630000903      쿠팡  20220630  공  
16                            타인을위한채무보증결정  20220630000901      쿠팡  20220630  공  
17                             특수관계인에대한출자  20220630000894      쿠팡  20220630  공  
18        대규모기업집단현황공시[연1회

In [ ]:
# http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20200414000766
html_text = dart.document('20200414000766')

df_list = pd.read_html(html_text)
len(df_list)

119

In [ ]:
# 연결포괄손익계산서
df_list[15]

과 목        주석   제 7 (당)기     제 6 (전)기
0               Ⅰ. 매출액  182324.0    7153094      4354586
1                 매출원가    1924.0    5950017      4142781
2             영업 일반관리비  192324.0    1923606      1339769
3              Ⅱ. 영업손실       NaN  (720,529)  (1,127,964)
4                 기타수익       NaN       1782          436
5                 기타비용       NaN       1481          666
6                 금융수익     520.0      50396        19338
7                 금융비용     520.0      54155        33450
8       Ⅲ. 법인세비용차감전순손실       NaN  (723,987)  (1,142,306)
9             Ⅳ. 법인세비용      14.0      (747)         2331
10            Ⅴ. 당기순손실       NaN  (723,240)  (1,144,637)
11          지배기업 소유주지분       NaN  (723,240)  (1,144,637)
12               비지배지분       NaN          -            -
13           Ⅵ. 기타포괄손실       NaN   (10,504)      (2,826)
14  당기손익으로 재분류되지 않는 항목       NaN   (10,504)      (2,826)
15      순확정급여부채의 재측정요소      13.0   (10,504)      (2,826)
16          Ⅶ. 당기총포괄손실       NaN  (733,744)  (1,147,463)
17          지배기업 소유주지분       NaN  (733,744)  (1,147,463)
18               비지배지분       NaN          -            -

당기(2019년)에 영업손실 -7,205 억원을 기록했습니다. 전기(2018년)에 -1조1279억원에 비하면 많은 개선이 이루어졌습니다.


***2019-2022 [FinanceData.KR]()***